### PR TITLE
[NOT NOW]: WIP graph to API v2

### PR DESCRIPTION
**Pull request by RobertJoonas**
The aim of this work was to make a start with migrating the dashboard to API v2, adding a new controller action for `POST /main-graph` that would use `Query.build` rather than `Query.from` (`Legacy.QueryBuilder`).

However, we've decided not to continue with this for now, due to several complications:

* APIv2 is currently missing functionality that would still have to be included in the controller:
  * Comparisons
  * Revenue goals
* There is no clear vision yet for how the frontend should make these requests (e.g. `POST` vs `GET`) and it makes more sense to work on migrating the whole dashboard to API v2 at once, taking into account the requirements of other endpoints, not just `main_graph`.

In conclusion, this is not something we want to rush into, and there's no real benefit in doing it at this stage.

Nevertheless, some of the code in this PR might become useful, so I'm keeping this PR open as a draft until necessary.

### CODE REVIEW

**Comment by macobo** [comment]
Let's close for now - made a note of this work and will dig it up when converting main_graph.
